src/esrs/state.rs

```diff
@@ -70,17 +90,27 @@ impl<S: Default> AggregateState<S> {
         &self.inner
     }
 
-    /// Returns the internal sequence number incremented by 1.
-    pub const fn next_sequence_number(&self) -> SequenceNumber {
+    /// Consumes self and extracts the internal state.
+    pub fn into_inner(self) -> S {
+        self.inner
+    }
+
+    /// Returns the internal sequence number.
+    pub const fn sequence_number(&self) -> &SequenceNumber {
+        &self.sequence_number
+    }
+
+    /// Computes the internal sequence number incremented by 1.
+    pub fn next_sequence_number(&self) -> SequenceNumber {
```

Comment:
Removed the `const` from this function. Nothing guarantees that, in the future, computing the next sequence number will be a const operation (it just happens to be at the moment).